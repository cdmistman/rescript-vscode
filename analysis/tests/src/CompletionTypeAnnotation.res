type someRecord = {
  age: int,
  name: string,
}

type someVariant = One | Two(bool)

type somePolyVariant = [#one | #two(bool)]

// let x: someRecord =
//                    ^com

// let x: someRecord = {}
//                      ^com

// let x: someVariant =
//                     ^com

// let x: someVariant = O
//                       ^com

// let x: somePolyVariant =
//                         ^com

// let x: somePolyVariant = #o
//                            ^com
