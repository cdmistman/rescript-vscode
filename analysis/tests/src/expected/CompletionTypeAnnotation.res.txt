Complete src/CompletionTypeAnnotation.res 9:22
XXX Not found!
Completable: Cexpression Type[someRecord]
[{
    "label": "{}",
    "kind": 4,
    "tags": [],
    "detail": "Inline record",
    "documentation": null,
    "sortText": "A",
    "insertText": "{$0}",
    "insertTextFormat": 2
  }]

Complete src/CompletionTypeAnnotation.res 12:24
XXX Not found!
Completable: Cexpression Type[someRecord]->recordBody
[{
    "label": "age",
    "kind": 4,
    "tags": [],
    "detail": "Inline record",
    "documentation": null
  }, {
    "label": "name",
    "kind": 4,
    "tags": [],
    "detail": "Inline record",
    "documentation": null
  }]

Complete src/CompletionTypeAnnotation.res 15:23
XXX Not found!
Completable: Cexpression Type[someVariant]
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\ntype  = One | Two(bool)",
    "documentation": null,
    "insertText": "One",
    "insertTextFormat": 2
  }, {
    "label": "Two(_)",
    "kind": 4,
    "tags": [],
    "detail": "Two(bool)\n\ntype  = One | Two(bool)",
    "documentation": null,
    "insertText": "Two(${1:_})",
    "insertTextFormat": 2
  }]

Complete src/CompletionTypeAnnotation.res 18:25
XXX Not found!
Completable: Cexpression Type[someVariant]=O
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\ntype  = One | Two(bool)",
    "documentation": null,
    "sortText": "A One",
    "insertText": "One",
    "insertTextFormat": 2
  }, {
    "label": "Obj",
    "kind": 9,
    "tags": [],
    "detail": "file module",
    "documentation": null
  }, {
    "label": "Object",
    "kind": 9,
    "tags": [],
    "detail": "file module",
    "documentation": null
  }]

Complete src/CompletionTypeAnnotation.res 21:27
XXX Not found!
Completable: Cexpression Type[somePolyVariant]
[{
    "label": "#one",
    "kind": 4,
    "tags": [],
    "detail": "#one\n\n[#one | #two(bool)]",
    "documentation": null,
    "insertText": "#one",
    "insertTextFormat": 2
  }, {
    "label": "#two(_)",
    "kind": 4,
    "tags": [],
    "detail": "#two(bool)\n\n[#one | #two(bool)]",
    "documentation": null,
    "insertText": "#two(${1:_})",
    "insertTextFormat": 2
  }]

Complete src/CompletionTypeAnnotation.res 24:30
XXX Not found!
Completable: Cexpression Type[somePolyVariant]=#o
[{
    "label": "#one",
    "kind": 4,
    "tags": [],
    "detail": "#one\n\n[#one | #two(bool)]",
    "documentation": null,
    "insertText": "one",
    "insertTextFormat": 2
  }]

